pkg/test/case.go

```diff
@@ -397,6 +397,11 @@ func (t *Case) determineNamespace() *namespace {
 	if t.PreferredNamespace == "" {
 		ns.Name = fmt.Sprintf("kuttl-test-%s", petname.Generate(2, "-"))
 		ns.AutoCreated = true
+	} else {
+		exist, _ := t.NamespaceExists(t.PreferredNamespace)
```

Comment:
Please do propagate and fail the test if the check fails due to bad cluster config, etc.